### DISCUSSION TITLE
Testing the AdaptiveDialogs

### DISCUSSION THREAD

**Discussion by Quintonvv** [mono0926/adaptive_dialog#126] [General]
I'm trying to add widget tests for some of my widgets using the adaptive_dialog package, but I can't seem to succeed in that. If I'm expecting an AlertDialog, I get a timer pending error, which is not on my side, it might not even be on your side either. Is there a way to test the widget?

The error I'm getting:
_The following assertion was thrown running a test:
A Timer is still pending even after the widget tree was disposed.
'package:flutter_test/src/binding.dart':
Failed assertion: line 1518 pos 12: '!timersPending'_

**Comment by mono0926** [comment]
You can write the test like this: https://github.com/mono0926/adaptive_dialog/blob/feature/mono/test/example/test/pages/alert_page_test.dart .

> The error I'm getting:
The following assertion was thrown running a test:
A Timer is still pending even after the widget tree was disposed.
'package:flutter_test/src/binding.dart':
Failed assertion: line 1518 pos 12: '!timersPending'

I think it's a problem with your code.
